site/docs/sql/intro.md

```diff
@@ -0,0 +1,306 @@
+need to create a Java class for the value:
+
+```java
+public class Trade implements Serializable {
+    public String ticker;
+    public BigDecimal price;
+    public long amount;
+}
+```
+
+We used public fields, but of course you can use private fields and use
+setters/getters. This class must be available to the cluster. You can
+either add it to the members class paths by creating a JAR file and
+adding to the `lib` folder, or you can use User Code Deployment. The
+user code deployment has to be enabled on the members; add the following
+section to the `config/hazelcast.yaml` file:
+
+```yaml
+hazelcast:
+  user-code-deployment:
+    enabled: true
+```
+
+Then use a client to upload the class:
+
+```java
+ClientConfig clientConfig = new JetClientConfig();
+clientConfig.getUserCodeDeploymentConfig()
+            .setEnabled(true)
+            .addClass(Trade.class);
+JetInstance jet = Jet.newJetClient(clientConfig);
+```
+
+After this, you can create the mapping for the IMap. The name of the
+IMap is `latest_trades`:
+
+```sql
+CREATE MAPPING latest_trades
+TYPE IMap
+OPTIONS (
+    keyFormat 'java',
+    keyJavaClass 'java.lang.String',
+    valueFormat 'java',
+    valueJavaClass 'com.example.Trade'
+)
+```
+
+Note that we omitted the column list in this query. It will be
+determined automatically according to the OPTIONS. Since we use `String`
+as the key class, the default column name for the key will be used:
+`__key`. The properties of the `Trade` class will be used. So the
+mapping will behave as if the following columns were specified in the
+previous statement:
+
+```sql
+(
+    __key VARCHAR,
+    ticker VARCHAR,
+    price DECIMAL,
+    amount BIGINT
+)
+```
+
+### Streaming Messages from the Kafka Topic to the IMap
+
+After creating the mapping for the `latest_trades` IMap, we can submit
+the following statement:
+
+```sql
+SINK INTO latest_trades(__key, ticker, price, amount)
+SELECT ticker, ticker, price, amount
+FROM trades
+```
+
+It will put the fields from the Kafka topic into the IMap, replicating
+the `ticker` twice: once into the value and once into the key. Since we
+use the `ticker` as the key, every new trade for the same ticker will
+overwrite the previous trade for that ticker, hence the name
+`latest_trades`.
+
+### Creating a Streaming Job
+
+However, you cannot directly execute the above command because it would
+never complete. Remember, it's reading from a Kafka topic, which is a
+stream of messages. Since it doesn't return any rows to the client, you
+must create a job for it:
+
+```sql
+CREATE JOB trades_ingestion AS
+SINK INTO latest_trades(__key, ticker, price, amount)
+SELECT ticker, ticker, price, amount
+FROM trades
+```
+
+The part after the `AS` keyword is same as the previous statement. The
+command will create a job named `trades_ingestion`. Now, even if the
+client disconnects, the cluster will continue running the job.
+
+### Cancelling the Job
+
+To cancel the job, use:
+
```

Comment:
`latest_trades` map can be queried using SQL. Please provide a sample command and link to IMDG SQL. This is a marketing touch to present the big picture.